usr/lib/ep11_stdll/new_host.c

```diff
@@ -2058,10 +2058,15 @@ CK_RV SC_EncryptInit(STDLL_TokData_t *tokdata, ST_SESSION_HANDLE *sSession,
         sess->encr_ctx.key = hKey;
 
         sess->encr_ctx.mech.mechanism = pMechanism->mechanism;
-        sess->encr_ctx.mech.pParameter = malloc(pMechanism->ulParameterLen);
-        memcpy(sess->encr_ctx.mech.pParameter, pMechanism->pParameter,
-               pMechanism->ulParameterLen);
-        sess->encr_ctx.mech.ulParameterLen = pMechanism->ulParameterLen;
+        if (pMechanism->pParameter && pMechanism->ulParameterLen > 0) {
+            sess->encr_ctx.mech.pParameter = malloc(pMechanism->ulParameterLen);
+            memcpy(sess->encr_ctx.mech.pParameter, pMechanism->pParameter,
```

Comment:
While you are at it, please also check if the malloc has worked, before attempting to memcpy.

Reply:
yep, that makes sense. Done.